LayoutTests/imported/w3c/web-platform-tests/import-maps/no-referencing-script-integrity-valid.html

```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+let log = [];
+</script>
+<script type="importmap">
+{
+  "integrity": {
+    "./resources/log.js?pipe=sub&name=NoReferencingScriptValidCheck": "sha384-5eRmXQSBE6H5ENdymdZxcyiIfJL1dxtH8p+hOelZY7Jzk+gt0gYyemrGY0cEaThF"
+  }
+}
+</script>
+<script>
+let promiseResolve;
```

Comment:
I'm still not loving this approach... putting the handler declaratively in the HTML makes it quite difficult to debug if things go wrong. Plus the inverted promise seems to add unnecessary complexity. 

Could the following work instead? it seems to achieve the same thing:

If we make the image just `<img>` and then:

```JS
promise_test(async () => {
  const img = document.querySelector("img");
  const promise = new Promise((resolve, reject) => {
    img.onload = () => {
      import('./resources/log.js?pipe=sub&name=NoReferencingScriptValidCheck').then(resolve).catch(reject);
    };
    img.src = "/images/green.png?2";
  });

  await promise;
  assert_equals(log.length, 1);
  assert_equals(log[0], "log:NoReferencingScriptValidCheck");
}, "Script was loaded as its valid integrity check passed");
```

Reply:
Even so, shouldn't it then include both the inline HTML handler and the IDL event handler attribute test, no?

Reply:
Sure, I can add such tests to dynamic-integrity (since an IDL event handler test would have a referencing script).